src/status_im/signing/core.cljs

```diff
@@ -0,0 +1,250 @@
+         {:signing.fx/sign-message {:params       {:data     data
+                                                   :password (security/safe-unmask-data password)
+                                                   :account  from}
+                                    :on-completed #(re-frame/dispatch [:signing/sign-message-completed %])}})))))
+
+(fx/defn send-transaction
+  {:events [:signing.ui/sign-is-pressed]}
+  [{{:signing/keys [sign tx] :as db} :db :as cofx}]
+  (let [{:keys [in-progress? password]} sign
+        {:keys [tx-obj gas gasPrice message]} tx]
+    (if message
+      (sign-message cofx)
+      (let [tx-obj-to-send (merge tx-obj
+                                  {:from (ethereum/current-address db)}
+                                  (when gas
+                                    {:gas (str "0x" (abi-spec/number-to-hex gas))})
+                                  (when gasPrice
+                                    {:gasPrice (str "0x" (abi-spec/number-to-hex gasPrice))}))]
+        (when-not in-progress?
+          {:db                          (update db :signing/sign assoc :error nil :in-progress? true)
+           :signing/send-transaction-fx {:tx-obj   tx-obj-to-send
+                                         :password password
+                                         :cb       #(re-frame/dispatch [:signing/transaction-completed % tx-obj-to-send])}})))))
+
+(fx/defn prepare-unconfirmed-transaction
+  [{:keys [db now]} hash {:keys [value gasPrice gas data to from]} symbol]
+  (let [all-tokens (:wallet/all-tokens db)
+        chain      (:chain db)
+        token      (tokens/symbol->token all-tokens (keyword chain) symbol)]
+    {:db (assoc-in db [:wallet :transactions hash]
+                   {:timestamp (str now)
+                    :to        to
+                    :from      from
+                    :type      :pending
+                    :hash      hash
+                    :data      data
+                    :token     token
+                    :symbol    symbol
+                    :value     (money/to-fixed (money/bignumber value))
+                    :gas-price (money/to-fixed (money/bignumber gasPrice))
+                    :gas-limit (money/to-fixed (money/bignumber gas))})}))
+
+(defn get-transfer-token [db to data]
+  (let [{:keys [symbol decimals] :as token} (tokens/address->token (:wallet/all-tokens db) (ethereum/chain-keyword db) to)]
+    (when (and token data (string? data) (string/starts-with? data (abi-spec/signature->method-id "transfer(address,uint256)")))
+      (let [[address value] (abi-spec/decode (str "0x" (subs data 10)) ["address" "uint256"])]
+        (when (and address value)
+          {:to       address
+           :contact  (get-contact db address)
+           :contract to
+           :amount   (money/to-fixed (money/token->unit value decimals))
+           :token    token
+           :symbol   symbol})))))
+
+(defn parse-tx-obj [db {:keys [to value data]}]
```

Comment:
I would prefer we really don't import this from the whole wallet this distinction between transaction and transaction-object was uncalled for and the use of abbreviations stands out with the rest of the codebase

Reply:
i agree here, but tbh haven't found a time to figure out how to do it better, so i decided to go with tx for signing screen, and tx-obj, raw transaction object to sign, yes we need to split it somehow and use better naming i agree